app/controllers/publish_controller.rb

```diff
@@ -55,6 +55,14 @@ def assign_autocomplete_warning
     @autocomplete_warning = AutocompleteItemsPresenter.new(service, service_autocomplete_items)
   end
 
+  def assign_from_address_presenter
+    @from_address_presenter = FromAddressPresenter.new(from_address)
```

Comment:
Could we not provide the `page` into the presenter here?  Then the presetner couls initialize the messages array with the right "page" of messages.  That way the view can just access `@from_address_presenter.message[text]`

This doesn't feel too icky to me, as the controller knows which 'page' it is presenting... so seems reasonabel to pass it in.  Could potentially even change `page` to `controller` / `controller_name` ?

Reply:
done!